src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/RSBridgePeripheral.java

```diff
@@ -0,0 +1,893 @@
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult isOnline() {
+        if (!isAvailable())
+            return notConnected();
+
+        return MethodResult.of(getComponent(EnergyNetworkComponent.class).getStored() > 0);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult getItem(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<ItemFilter, String> filter = ItemFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        ItemFilter parsedFilter = filter.getLeft();
+        if (parsedFilter.isEmpty())
+            return MethodResult.of(null, "EMPTY_FILTER");
+
+        Map<?, ?> resourceProperties = RsApi.getParsedItem(getNetwork(), parsedFilter);
+        if (resourceProperties == null)
+            return MethodResult.of(null, "NOT_FOUND");
+
+        return MethodResult.of(resourceProperties);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult getFluid(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<FluidFilter, String> filter = FluidFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        FluidFilter parsedFilter = filter.getLeft();
+        if (parsedFilter.isEmpty())
+            return MethodResult.of(null, "EMPTY_FILTER");
+
+        Map<?, ?> resourceProperties = RsApi.getParsedFluid(getNetwork(), parsedFilter);
+        if (resourceProperties == null)
+            return MethodResult.of(null, "NOT_FOUND");
+
+        return MethodResult.of(resourceProperties);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult getChemical(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<ChemicalFilter, String> filter = ChemicalFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        ChemicalFilter parsedFilter = filter.getLeft();
+        if (parsedFilter.isEmpty())
+            return MethodResult.of(null, "EMPTY_FILTER");
+
+        Map<?, ?> resourceProperties = RsApi.getParsedChemical(getNetwork(), parsedFilter);
+        if (resourceProperties == null)
+            return MethodResult.of(null, "NOT_FOUND");
+
+        return MethodResult.of(resourceProperties);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult listItems(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<ItemFilter, String> filter = ItemFilter.parse(arguments.optTable(0, Map.of()));
```

Comment:
will [`Collections.emptyMap()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Collections.html#emptyMap()) be better at here? so it doesn't create another map instance. Although its not a major impact.